src/operator/regression_output-inl.h

```diff
@@ -69,11 +69,11 @@ class RegressionOutputOp : public Operator {
     Stream<xpu> *s = ctx.get_stream<xpu>();
     real_t num_output =
       in_data[reg_enum::kLabel].Size()/in_data[reg_enum::kLabel].shape_[0];
-    Tensor<xpu, 2> out = out_data[reg_enum::kOut].FlatTo2D<xpu, real_t>(s);
-    Tensor<xpu, 2> grad = in_grad[reg_enum::kData].FlatTo2D<xpu, real_t>(s);
-    Tensor<xpu, 2> label = in_data[reg_enum::kLabel]
-      .get_with_shape<xpu, 2, real_t>(out.shape_, s);
-    Assign(grad, req[reg_enum::kData], param_.grad_scale/num_output*
+    Tensor<xpu, 2, DType> out = out_data[reg_enum::kOut].FlatTo2D<xpu, DType>(s);
+    Tensor<xpu, 2, DType> grad = in_grad[reg_enum::kData].FlatTo2D<xpu, DType>(s);
+    Tensor<xpu, 2, DType> label = in_data[reg_enum::kLabel]
+      .get_with_shape<xpu, 2, DType>(out.shape_, s);
+    Assign(grad, req[reg_enum::kData], scalar<DType>(param_.grad_scale/num_output)*
```

Comment:
Thank you for your review. Which line are you worry about? I think int8 is currently not supported as we are using `MSHADOW_REAL_TYPE_SWITCH` to create operators.